lib/api/contacts.rb

```diff
@@ -12,14 +12,10 @@ def hash_access
     end
 
     def params
-      super
-    end
-
-    def opts
-      '&property=vid&property=firstname&property=lastname&'\
+      super({property: '&property=vid&property=firstname&property=lastname&'\
```

Comment:
super refers to the parent which in this case is Base.  Base doesn't have a opts method.  Also since URI doesn't allow duplicate keys we have to add this to the end of the URI sting. which happens on line 15 of read and parse: 

```ruby
# this helps atleast with the contacts url by appending the
      # string of multiple '&property='
      uri.query += opts if !opts.empty?